test/services/next_references/next_transaction_file_reference.service.test.js

```diff
@@ -26,43 +26,47 @@ describe('Next Transaction File Reference service', () => {
 
   describe('When a valid region and regime are specified', () => {
     it('returns a correctly formatted file reference', async () => {
-      const result = await NextTransactionFileReferenceService.go(regime, 'R')
+      const result = await NextTransactionFileReferenceService.go(regime, 'R', 'presroc')
 
       expect(result).to.equal('nalri50001')
     })
 
     describe('the file reference generated', () => {
       it('increments with each call', async () => {
-        const result = await NextTransactionFileReferenceService.go(regime, 'R')
-        const secondResult = await NextTransactionFileReferenceService.go(regime, 'R')
+        const result = await NextTransactionFileReferenceService.go(regime, 'R', 'presroc')
+        const secondResult = await NextTransactionFileReferenceService.go(regime, 'R', 'presroc')
 
-        // The call to slice(-1) grabs the last character from the returned string
-        expect(result.slice(-1)).to.equal('1')
-        expect(secondResult.slice(-1)).to.equal('2')
+        expect(result).endsWith('1')
+        expect(secondResult).endsWith('2')
       })
 
-      it('increments with each call independently for each regime & region', async () => {
+      it('increments with each call independently for each regime, region and ruleset', async () => {
         const otherRegime = await RegimeHelper.addRegime('cfd', 'CFD')
         await SequenceCounterHelper.addSequenceCounter(otherRegime.id, 'S')
 
-        const result = await NextTransactionFileReferenceService.go(regime, 'R')
+        const result = await NextTransactionFileReferenceService.go(regime, 'R', 'presroc')
         const otherResult = await NextTransactionFileReferenceService.go(otherRegime, 'S')
 
-        // The call to slice(-1) grabs the last character from the returned string
-        expect(result.slice(-1)).to.equal('1')
-        expect(otherResult.slice(-1)).to.equal('1')
+        expect(result).endsWith('1')
+        expect(otherResult).endsWith('1')
       })
 
       it('has a prefix specific to the regime', async () => {
         const otherRegime = await RegimeHelper.addRegime('cfd', 'CFD')
         await SequenceCounterHelper.addSequenceCounter(otherRegime.id, 'S')
 
-        const result = await NextTransactionFileReferenceService.go(regime, 'R')
+        const result = await NextTransactionFileReferenceService.go(regime, 'R', 'presroc')
         const otherResult = await NextTransactionFileReferenceService.go(otherRegime, 'S')
 
         expect(result).startsWith('nal')
         expect(otherResult).startsWith('cfd')
       })
+
+      it('has the correct suffix for a ruleset', async () => {
+        const result = await NextTransactionFileReferenceService.go(regime, 'R', 'sroc')
```

Comment:
I think it wouldn't hurt to have a test that checks presroc results in no suffix. This would mainly be for documentation purposes. The test would then hopefully make it clear what to expect in both situations.

What do you think?

Reply:
Agreed, I've expanded the test to check for both the expected sroc and presroc filename endings.